packages/common/src/content/contentUtils.ts

```diff
@@ -437,3 +438,69 @@ const getContentRelativeUrl = (
     content.typeKeywords
   );
 };
+
+export interface IGetServiceStatusOptions extends IGetLayerOptions {
+  timeout?: number;
+}
+
+// Tests can be found in packages/common/test/content/content.test.ts
+const unavailable = {
+  kind: "service",
+  service: {
+    availability: "unavailable",
+  },
+} as IHubServiceBackedContentStatus;
+const slow = {
+  kind: "service",
+  service: {
+    availability: "slow",
+  },
+} as IHubServiceBackedContentStatus;
+const available = {
+  kind: "service",
+  service: {
+    availability: "available",
+  },
+} as IHubServiceBackedContentStatus;
+
+/**
+ * Get the status of a content item
+ * @param entity the content item
+ * @returns the status of the content item
+ */
+export async function getServiceStatus(
+  entity: IHubEditableContent,
+  options: IGetServiceStatusOptions
+): Promise<IHubServiceBackedContentStatus> {
+  // get the request options for the `getService` call, and set a default timeout if one is not provided
+  const { timeout = 3000, ...requestOptions } = options;
+
+  if (entity.url) {
```

Comment:
As outcome of current [opendata-ui PR](https://github.com/ArcGIS/opendata-ui/pull/13855) to address non-service backed items, we should add this fix here as well.

Reply:
I think I've made all the updates needed. @tomwayson, can you confirm this PR looks good to you before I merge?